### PR TITLE
Add a sidecar container to reload user cluster Prometheus configuration

### LINKED ISSUE #7486: User Cluster Prometheus: Automate configuration reloading
Make user cluster Prometheus reload configuration automatically by adding a sidecar container, which will curl the reload endpoint.

### PR DESCRIPTION
**Pull request by jiachengxu**
**What this PR does / why we need it**:
This PR adds a sidecar container to the user cluster Prometheus to reload configuration automatically.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7486 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/user-cluster-controller-manager/resources/resources/mla/prometheus/deployment.go [hunk 1]
```diff
@@ -133,6 +137,49 @@ func DeploymentCreator() reconciling.NamedDeploymentCreatorGetter {
...
 				},
+				{
+					Name:            "prometheus-config-reloader",
+					Image:           fmt.Sprintf("%s/%s:%s", resources.RegistryQuay, reloaderImageName, reloaderTag),
+					ImagePullPolicy: corev1.PullAlways,
+					Args: []string{
+						// Fully usage of prometheus-config-reloader:
+						// https://github.com/prometheus-operator/prometheus-operator/blob/v0.49.0/cmd/prometheus-config-reloader/main.go#L72-L108
+						"--listen-address=:8080",
+						"--watch-interval=10s",
```
Comment (jiachengxu) [inline pkg/controller/user-cluster-controller-manager/resources/resources/mla/prometheus/deployment.go:158]: @rastislavs From https://github.com/prometheus-operator/prometheus-operator/blob/v0.49.0/cmd/prometheus-config-reloader/main.go#L79, 
`watch-interval` means "how often the reloader re-reads the configuration file and directories", do you think re-reading/checking configuration every 10s is sufficient?

Comment (rastislavs) [reply pkg/controller/user-cluster-controller-manager/resources/resources/mla/prometheus/deployment.go:158]: yeah, 10s is good enough IMO

**Pull request merged**